driver/driver.go

```diff
@@ -29,6 +29,9 @@ type Driver interface {
 	// UpdateTask supports updating certain fields of a task
 	UpdateTask(ctx context.Context, task PatchTask) (InspectPlan, error)
 
+	// ValidateTask validates the configurations of the task
+	ValidateTask(ctx context.Context) error
+
```

Comment:
I was wondering what your thoughts were if validation was instead happened within InitTask() instead of in its own method? No opinion, just want to understand better! Also curious @findkim of your thoughts since you've been doing some refactoring of the driver interface lately.